tests/conftest.py

```diff
@@ -0,0 +1,206 @@
+"""Fixture-sharing file for test suite.
```

Comment:
I'd prefer the tests to be in `cyipopt/tests` so it is part of the cyipopt package.

Reply:
What's your reasoning for wanting the tests distributed as part of the package? I agree that having the tests inlined with the package can make sense when you have a lot of unit tests that map directory to the module structure, however the "test suite" here is predominantly integration tests so that is not the case.

The tests are for the benefit of developers. We have the `examples/` directory which the user can use to verify the package is working, and as most will install using _conda_ it's highly unlikely that they'd need the tests themselves. If we want the tests distributed with the package then we can always use `MANIFEST.ini`.

I'm very happy to refactor the tests back under `cyipopt/` though if you'd prefer :)

Reply:
I always include the tests so that they are subpackages so that anyone who installs the package can run the tests easily. One use case is that someone reports a bug and you ask them to run the test suite and report the output. If the tests are installed with the package, then something a simple as:

```python
import ipopt
ipopt.test()
```

can be run. If they aren't included they have to download the tarball (which might have the tests) or clone the git repo to get access to the tests. The tests aren't just for developers. They can be useful to users and are often useful to downstream packagers.

For the purposes of conda forge it is much preferable to have the tests in the package if you want to run one or more of them to verify the package works. Conda forge installs the package into an isolated test environment and if the tests aren't installed with the package you have to do more work to make them accessible.

As long as the tests aren't increasing the size of the tarball by large #'s of mb, I don't know of a reason *not* to include them.

Here is a discussion I had on pelican last year:

https://github.com/getpelican/pelican/issues/2635

about the same issue. There were extra hoops to go through to package pelican with conda forge because the tests were missing.

All of the major scientific python packages seem to install the tests and typically offer, e.g., `sympy.test()` or `numpy.test()`.

If this PR introduces unit tests of any kind, I'd prefer that they are installed as a subpackage to cyipopt.

Reply:
I believe `MANIFEST.in` is only used to declare what is in the the sdist, not what is ultimately installed.

Reply:
Thanks very much for the detail @moorepants. That all makes sense and is a better justification than my argument so I'll now agree that keeping the tests as an installed sub package is a good idea. I'll refactor as per your suggestion. Thanks! :)

Reply:
For the time being I'm going to move all the tests back to `cyipopt/tests/` but I will not be implementing what you suggested above

```python
import cyipopt
cyipopt.test()
```

as I need to do a bit more looking at other projects (e.g. _Sympy_) to see how this is implemented. I'd assume doing so would add a mandatory dependency on _pytest_ (which is maybe what we want if we're distributing the tests as part of the package itself)? For the time being the tests will have to be run explicitly using _pytest_ and we can open an issue for the above `cyipopt.test()` function if we decide we want it.

Reply:
Addressed in https://github.com/matthias-k/cyipopt/pull/75/commits/140ecab94ceca6958a54e827cf0adad5336ee7d7 and https://github.com/matthias-k/cyipopt/pull/75/commits/784f633bd0e8506781fa30c865a4157c843f066e.

Reply:
Putting the tests in `cyipopt/tests` with a `cyipopt/tests/__init__.py` is sufficient.

Reply:
As for the py.test dependency, we could simply raise an importerror if it isn't installed, like we do with scipy.